src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java

```diff
@@ -41,10 +43,22 @@ public void ifAcquiredOrElse(BlobClient blobClient, Runnable onSuccess, Runnable
                 );
             }
 
-            onFailure.run();
+            if (exc.getErrorCode() == BLOB_NOT_FOUND) {
+                onBlobNotFound.run();
+            } else {
+                logger.info(
+                    "Cannot acquire a lease for blob - skipping. File name: {}, container: {}",
+                    blobClient.getBlobName(),
+                    blobClient.getContainerName()
+                );
+            }
```

Comment:
Should this change be in this PR?
It mentions enabling using lease id.

Reply:
it also mentions `Refactor`. It is basic look over. For following will have to handle `BLOB_NOT_FOUND` individually so ending up with 3 functional interface params in method. My gut feeling told team would not like it hence refactor a bit more. Very minimal tbh